### PR TITLE
refactor: add constant subcategory for all resources and data sources

### PR DESCRIPTION
**Pull request by gaetanars**
### Description of your changes
Add SubCategory for all resources

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Write or modify templates in `templates/` directory
- [x] Run `make generate` to ensure the doc was updated properly

### CODE REVIEW

**Review by dmicheneau — DISMISSED** [review]

File: docs/data-sources/edgegateway.md [hunk 1]
```diff
@@ -1,16 +1,25 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
-page_title: "cloudavenue_edge_gateway Data Source - cloudavenue"
-subcategory: ""
+page_title: "cloudavenue_edgegateway Data Source - cloudavenue"
+subcategory: "Edge Gateway (Tier-1)"
```
Comment (dmicheneau) [inline docs/data-sources/edgegateway.md:3]: precise this in description (Tier-1)

File: docs/data-sources/edgegateways.md [hunk 2]
```diff
@@ -1,22 +1,21 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
-page_title: "cloudavenue_edge_gateways Data Source - cloudavenue"
-subcategory: ""
+page_title: "cloudavenue_edgegateways Data Source - cloudavenue"
+subcategory: "Edge Gateway (Tier-1)"
```
Comment (dmicheneau) [inline docs/data-sources/edgegateways.md:3]: precise this in description (Tier-1)

File: docs/data-sources/iam_group.md [hunk 3]
```diff
@@ -1,19 +1,18 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
-page_title: "cloudavenue_org_group Data Source - cloudavenue"
-subcategory: ""
+page_title: "cloudavenue_iam_group Data Source - cloudavenue"
+subcategory: "IAM (Identity & Access Management)"
```
Comment (dmicheneau) [inline docs/data-sources/iam_group.md:3]: precise this in description (Identity & Access Management)

File: docs/data-sources/iam_role.md [hunk 4]
```diff
@@ -1,19 +1,18 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
-page_title: "cloudavenue_org_role Data Source - cloudavenue"
-subcategory: ""
+page_title: "cloudavenue_iam_role Data Source - cloudavenue"
+subcategory: "IAM (Identity & Access Management)"
```
Comment (dmicheneau) [inline docs/data-sources/iam_role.md:3]: precise this in description (Identity & Access Management)

File: docs/data-sources/iam_user.md [hunk 5]
```diff
@@ -1,24 +1,23 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
-page_title: "cloudavenue_org_user Data Source - cloudavenue"
-subcategory: ""
+page_title: "cloudavenue_iam_user Data Source - cloudavenue"
+subcategory: "IAM (Identity & Access Management)"
```
Comment (dmicheneau) [inline docs/data-sources/iam_user.md:3]: precise this in description (Identity & Access Management)

File: docs/data-sources/tier0_vrf.md [hunk 6]
```diff
@@ -1,7 +1,6 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
 page_title: "cloudavenue_tier0_vrf Data Source - cloudavenue"
-subcategory: ""
+subcategory: "tier-0 VRF Gateways"
```
Comment (dmicheneau) [inline docs/data-sources/tier0_vrf.md:3]: it's not a network subcategory ?

File: docs/data-sources/tier0_vrfs.md [hunk 7]
```diff
@@ -1,7 +1,6 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
 page_title: "cloudavenue_tier0_vrfs Data Source - cloudavenue"
-subcategory: ""
+subcategory: "tier-0 VRF Gateways"
```
Comment (dmicheneau) [inline docs/data-sources/tier0_vrfs.md:3]: it's not a network subcategory ?

File: docs/data-sources/vapp.md [hunk 8]
```diff
@@ -1,7 +1,6 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
 page_title: "cloudavenue_vapp Data Source - cloudavenue"
-subcategory: ""
+subcategory: "vApp (Virtual Appliance)"
```
Comment (dmicheneau) [inline docs/data-sources/vapp.md:3]: put "(Virtual Appliance)" in Description

File: docs/data-sources/vdc.md [hunk 9]
```diff
@@ -1,7 +1,6 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
 page_title: "cloudavenue_vdc Data Source - cloudavenue"
-subcategory: ""
+subcategory: "vDC (Virtual Datacenter)"
```
Comment (dmicheneau) [inline docs/data-sources/vdc.md:3]: put in (Virtual Datacenter) in Description

File: docs/data-sources/vdcs.md [hunk 10]
```diff
@@ -1,7 +1,6 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
 page_title: "cloudavenue_vdcs Data Source - cloudavenue"
-subcategory: ""
+subcategory: "vDC (Virtual Datacenter)"
```
Comment (dmicheneau) [inline docs/data-sources/vdcs.md:3]: (Virtual Datacenter) in description

File: docs/resources/edgegateway.md [hunk 11]
```diff
@@ -1,12 +1,11 @@
 ---
-# generated by https://github.com/hashicorp/terraform-plugin-docs
-page_title: "cloudavenue_edge_gateway Resource - cloudavenue"
-subcategory: ""
+page_title: "cloudavenue_edgegateway Resource - cloudavenue"
+subcategory: "Edge Gateway (Tier-1)"
```
Comment (dmicheneau) [inline docs/resources/edgegateway.md:3]: move to desc

File: docs/resources/iam_group.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
+---
+page_title: "cloudavenue_iam_group Resource - cloudavenue"
+subcategory: "IAM (Identity & Access Management)"
```
Comment (dmicheneau) [inline docs/resources/iam_group.md:3]: in desc

File: docs/resources/vm_disk.md [hunk 13]
```diff
@@ -1,21 +1,20 @@
...
 
 ## Example Usage
 
 ```terraform
-resource "cloudavenue_vapp_vm_internal_disk" "example" {
-  vapp_name       = "vapp_test3"
-  vm_name         = "TestRomain"
+resource "cloudavenue_vm_disk" "example" {
+	vapp_name       = "vapp_test3"
+	vm_name         = "TestRomain"
```
Comment (dmicheneau) [inline docs/resources/vm_disk.md:17]: May be use another vm_name like = "ExampleVMName"

File: examples/data-sources/cloudavenue_edgegateways/data-source.tf [hunk 14]
```diff
@@ -0,0 +1,5 @@
+data "cloudavenue_edgegateways" "example" {}
+
+output "gateways" {
```
Comment (dmicheneau) [inline examples/data-sources/cloudavenue_edgegateways/data-source.tf:3]: For more visibility i suggest: output "list_of_gateways"

File: examples/resources/cloudavenue_/import.sh [hunk 15]
```diff
@@ -0,0 +1,2 @@
+# use the name to import the resource
+terraform import cloudavenue_.example name
```
Comment (dmicheneau) [inline examples/resources/cloudavenue_/import.sh:2]: An error .? the file name is incomplete, show path: "examples/resources/cloudavenue_"

File: examples/resources/cloudavenue_iam_user/import.sh [hunk 16]
```diff
@@ -0,0 +1,2 @@
+# use the user-name to import the User
+terraform import cloudavenue_iam_user.example exampleuser
```
Comment (dmicheneau) [inline examples/resources/cloudavenue_iam_user/import.sh:2]: why you don't use name like other example of import ?

terraform import cloudavenue_iam_user.example name

File: examples/resources/cloudavenue_vm_disk/resource.tf [hunk 17]
```diff
@@ -0,0 +1,11 @@
+resource "cloudavenue_vm_disk" "example" {
+	vapp_name       = "vapp_test3"
+	vm_name         = "TestRomain"
```
Comment (dmicheneau) [inline examples/resources/cloudavenue_vm_disk/resource.tf:3]: Use a better generic appropriate vapp_name and vm_name

**Review by dmicheneau — CHANGES_REQUESTED** [review]
See comments

**Review by eremid — APPROVED** [review]

**Pull request merged**